### PR TITLE
Features/civ 5611 submit lip claim

### PR DESCRIPTION
**Pull request by TatsyMaslova**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-5611

### Change description ###
client for submitting claim. 
This Pr contains some work from translation story https://tools.hmcts.net/jira/browse/CIV-5341 that is in blocked

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/services/features/claim/submission/submitClaim.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('partialAdmissionService');
+
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+export const submitClaim = async (req: AppRequest): Promise<Claim> => {
+  try {
+    const claim = await getCaseDataFromStore(req.session?.user?.id);
+    const ccdClaim = translateDraftClaimToCCD(claim);
```
Comment (TatsyMaslova) [inline src/main/services/features/claim/submission/submitClaim.ts:17]: this is a change from a dependant story that Kadir was working on

File: src/main/services/translation/claim/ccdTranslation.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {CCDClaim} from 'common/models/civilClaimResponse';
+import {Claim} from 'common/models/claim';
+import {toCCDParty} from '../response/convertToCCDParty';
+import {YesNoUpperCamelCase} from 'form/models/yesNo';
+
+export const translateDraftClaimToCCD = (claim: Claim): CCDClaim => {
+  return {
+    applicant1Represented: YesNoUpperCamelCase.NO,
+    applicant1: toCCDParty(claim.applicant1),
```
Comment (TatsyMaslova) [inline src/main/services/translation/claim/ccdTranslation.ts:9]: applicant1 and respondent1 field translations are from a dependant story that Kadir is working on

File: src/main/common/models/claim.ts [hunk 3]
```diff
@@ -87,8 +87,8 @@ export class Claim {
   public static fromCCDCaseData(ccdClaim: CCDClaim): Claim {
 
     const claim: Claim = Object.assign(new Claim(), ccdClaim);
-    claim.applicant1 = toCUIParty(ccdClaim.applicant1);
-    claim.respondent1 = toCUIParty(ccdClaim.respondent1);
+    claim.applicant1 = toCUIParty(ccdClaim?.applicant1);
```
Comment (laarmada) [inline src/main/common/models/claim.ts:90]: if we have a claim in ccd, it´s mandatory to have a applicant and respondent... no?

File: src/main/services/translation/response/convertToCCDParty.ts [hunk 4]
```diff
@@ -6,15 +6,15 @@ import {toCCDAddress} from './convertToCCDAddress';
 export const toCCDParty = (party: Party): CCDParty => {
   return {
     companyName: party?.type === PartyType.COMPANY ? party?.partyDetails?.partyName : undefined,
-    individualDateOfBirth: party?.type === PartyType.INDIVIDUAL ? party?.dateOfBirth?.date.toLocaleDateString() : undefined,
+    individualDateOfBirth: party?.type === PartyType.INDIVIDUAL ? party?.dateOfBirth?.date?.toLocaleDateString() : undefined,
```
Comment (laarmada) [inline src/main/services/translation/response/convertToCCDParty.ts:9]: that is changed in a branch pending to merge in Master.

File: src/test/unit/routes/features/claim/checkAnswersController.test.ts [hunk 5]
```diff
@@ -1,32 +1,25 @@
...
-import {getSummarySections} from '../../../../../main/services/features/claim/checkAnswers/checkAnswersService';
-import {CLAIM_CHECK_ANSWERS_URL} from '../../../../../main/routes/urls';
+import {getSummarySections} from 'services/features/claim/checkAnswers/checkAnswersService';
+import {CLAIM_CHECK_ANSWERS_URL, CLAIM_CONFIRMATION_URL} from 'routes/urls';
 import {TestMessages} from '../../../../utils/errorMessageTestConstants';
-import {getElementsByXPath} from '../../../../utils/xpathExtractor';
-import {createClaimWithBasicDetails} from '../../../../utils/mocks/claimDetailsMock';
+import request from 'supertest';
 
-const jsdom = require('jsdom');
```
Comment (laarmada) [inline src/test/unit/routes/features/claim/checkAnswersController.test.ts:9]: if that throw a time out we should remove also from response/checkAnswersController.test.ts

File: src/test/unit/routes/features/claim/checkAnswersController.test.ts [hunk 6]
```diff
@@ -41,104 +34,35 @@ describe('Response - Check answers', () => {
...
-        "//dd[@class='govuk-summary-list__value' and preceding-sibling::dt[contains(text(),'Email')]]",
-        htmlDocument);
-
-      expect(header.length).toBe(1);
-      expect(header[0].textContent).toBe(checkYourAnswerEng);
-      expect(fullName.length).toBe(2);
-      expect(fullName[0].textContent?.trim()).toBe(PARTY_NAME);
-      expect(fullName[1].textContent?.trim()).toBe(PARTY_NAME);
-      expect(address.length).toBe(2);
-      expect(address[0].textContent?.trim()).toBe('54 avenue');
```
Comment (laarmada) [inline src/test/unit/routes/features/claim/checkAnswersController.test.ts:94]: so we need some functionality test now we are not testing the views....

File: src/main/services/translation/response/convertToCUI/convertToCUIParty.ts [hunk 7]
```diff
@@ -10,27 +10,27 @@ import {CitizenDate} from '../../../../common/form/models/claim/claimant/citizen
 export const toCUIParty = (ccdParty: CCDParty): Party => {
   const cuiParty = new Party();
   cuiParty.partyDetails = new PartyDetails({});
-  if (ccdParty.type === PartyType.INDIVIDUAL) {
+  if (ccdParty?.type === PartyType.INDIVIDUAL) {
```
[the remark this answers is not in the document]
Comment (TatsyMaslova) [reply src/main/services/translation/response/convertToCUI/convertToCUIParty.ts:13]: it is not guaranteed so if ccd returns nothing then there will be an error in this function. Hence it has to make sure it handles this gracefully

File: src/test/unit/routes/features/claim/checkAnswersController.test.ts (hunk 6 above)
Comment (TatsyMaslova) [reply src/test/unit/routes/features/claim/checkAnswersController.test.ts:94]: we will test the views in the funtional tests that Raja and super team of testers are creating. if we have view tests in unit tests it makes them long and causes timeout. I think Maro has discussed this with platops

File: src/main/common/models/claim.ts (hunk 3 above)
Comment (TatsyMaslova) [reply src/main/common/models/claim.ts:90]: No we should not assume that ccdClaim is defined. We cannot guarantee that civil service will return case_data. we need to handle this gracefully. the issue was highlighted when i was running tests.

File: src/test/unit/routes/features/claim/checkAnswersController.test.ts (hunk 5 above)
Comment (TatsyMaslova) [reply src/test/unit/routes/features/claim/checkAnswersController.test.ts:9]: i only removed those tests that were related to the view. but left or amended the tests that actually test the controller

**Review by laperla11 — APPROVED** [review]

File: src/main/services/translation/response/convertToCCDParty.ts (hunk 4 above)
Comment (TatsyMaslova) [reply src/main/services/translation/response/convertToCCDParty.ts:9]: its ok - if it is changed once it is in master i can merge it in

File: src/test/unit/routes/features/claim/checkAnswersController.test.ts (hunk 5 above)
Comment (laarmada) [reply src/test/unit/routes/features/claim/checkAnswersController.test.ts:9]: I mean we have the same test in the response/checkAnswersController.test.ts

**Review by laarmada — APPROVED** [review]

**Comment by TatsyMaslova** [comment]
Best to remove it because it can cause timeout.. perhaps in a tech debt ticket

**Comment by TatsyMaslova** [comment]
reopen for deployment

**Review by akriti-hmcts — APPROVED** [review]

**Pull request merged**